lib/pointer.ts

```diff
@@ -120,10 +122,23 @@ class Pointer<S extends object = JSONSchema, O extends ParserOptions<S> = Parser
         }
 
         this.value = null;
-        throw new MissingPointerError(token, decodeURI(this.originalPath));
+
+        let path: any = '';
+
+        if (path !== undefined) {
```

Comment:
path will never be undefined since you define it two lines above? Also please dont use `: any`